src/server/pipeline.py

```diff
@@ -317,12 +350,88 @@ async def handle_message(
             yield connection, outbound_message, global_connections
             return
         elif full_message.function == "pong":
+            if connection.connection_type == NodeType.FULL_NODE:
+                await global_connections.update_connection_time(
+                    connection.get_peer_info()
+                )
             return
 
-        f_with_peer_name = getattr(api, full_message.function + "_with_peer_name", None)
+        # Peer gossip message handle.
+        if full_message.function == "request_peers":
```

Comment:
Absolutely not. The pipeline should not respond to messages. Rethink.

Reply:
Can you explain why? I don't think it should go in full_node.py either though

Reply:
I'm fine with it not going in full_node, but this is not the place for it. A connection manager-type thing which either attaches a listener to the end of the pipeline can do this.

The pipeline is supposed to do one thing: accept newly created sockets in one end and yield Chia messages out the other end (it probably already does too much, but this is all it's actually supposed to do). Jamming connection manager hooks into the middle of the pipeline will just make this already frightening code even less maintainable.

If messages that come down the pipe are not all for the full node object, then we're doing multiplexing. If we want multiplexing, do it explicitly. Add a multiplexing layer.